### PR TITLE
[MAINTENANCE] Refactor `DataContext.__init__` to move Cloud-specific logic to `CloudDataContext`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Port over logic relevant to Cloud to `CloudDataContext`

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/data_context/data_context/cloud_data_context.py [hunk 1]
```diff
@@ -23,13 +24,21 @@
...
     from great_expectations.checkpoint.checkpoint import Checkpoint
 
 logger = logging.getLogger(__name__)
 
 
+class GECloudEnvironmentVariable(str, Enum):
+    BASE_URL = "base_url"
+    ACCOUNT_ID = "account_id"  # Deprecated
+    ORGANIZATION_ID = "organization_id"
+    ACCESS_TOKEN = "access_token"
```
Comment (cdkini) [inline great_expectations/data_context/data_context/cloud_data_context.py:42]: New enum so we don't have magic strings

File: great_expectations/data_context/data_context/cloud_data_context.py [hunk 2]
```diff
@@ -406,3 +415,93 @@ def add_checkpoint(
             checkpoint_config=checkpoint_config, data_context=self  # type: ignore[arg-type]
         )
         return checkpoint
+
+    # TODO: deprecate ge_cloud_account_id
+    @classmethod
+    def _get_ge_cloud_config_dict(
```
Comment (cdkini) [inline great_expectations/data_context/data_context/cloud_data_context.py]: Both these methods used to live on `DataContext`

File: great_expectations/data_context/data_context/cloud_data_context.py [hunk 3]
```diff
@@ -406,3 +415,93 @@ def add_checkpoint(
...
+
+        # TODO: remove if/else block when ge_cloud_account_id is deprecated.
+        if ge_cloud_account_id is not None:
+            logger.warning(
+                'The "ge_cloud_account_id" argument has been renamed "ge_cloud_organization_id" and will be '
+                "deprecated in the next major release."
+            )
+        else:
+            ge_cloud_account_id = CloudDataContext._get_global_config_value(
+                environment_variable=GECloudEnvironmentVariable.ACCOUNT_ID,
```
Comment (cdkini) [inline great_expectations/data_context/data_context/cloud_data_context.py]: Only change here is the use of the enum

File: great_expectations/data_context/data_context/data_context.py [hunk 4]
```diff
@@ -227,124 +226,69 @@ def scaffold_custom_data_docs(cls, plugins_dir: str) -> None:
...
-            )
-            or "https://app.greatexpectations.io/"
+    ) -> None:
+        self._ge_cloud_mode = ge_cloud_mode
+        self._ge_cloud_config = self._init_ge_cloud_config(
+            ge_cloud_mode=ge_cloud_mode,
+            ge_cloud_base_url=ge_cloud_base_url,
+            ge_cloud_account_id=ge_cloud_account_id,
+            ge_cloud_access_token=ge_cloud_access_token,
+            ge_cloud_organization_id=ge_cloud_organization_id,
```
Comment (cdkini) [inline great_expectations/data_context/data_context/data_context.py:243]: two new helpers:
- `_init_ge_cloud_config`
- `_init_context_root_directory`

**Review by anthonyburdi — APPROVED** [review]
LGTM! 🙇

**Pull request merged**